draft-ietf-masque-connect-udp-listen.md

```diff
@@ -188,6 +188,10 @@ validate the target on every datagram and MUST NOT forward individual datagrams
 with unauthorized targets. Proxies can either silently discard such datagrams or
 abort the corresponding request stream.
 
+In addition, Proxies SHOULD consider restricting the IPs allowed to connect
+to the client via the proxy and reject any unauthorized connections on the
+target side, in order to prevent the misuse of the proxy.
```

Comment:
It is not clear to me how the proxy would know which IPs are safe and which ones aren't.

Reply:
Determining what qualifies as a safe or accepted target is an exercise that should be left to the proxy implementation in my opinion. This applies both ways, the proxy may only want to provide its proxying capability to certain targets, and likewise only let certain targets talk to the client, or they allow all traffic either way, knowing the risks involved, up to them to monitor or limit the traffic.

Reply:
Unfortunately, leaving this as exercise to the proxy leads to interoperability issues. In this case, one proxy might decide to restrict to addresses that the client has sent to, whereas another might restrict to address and ports. Now clients don't know which is which and have to guess. It's almost always better to be explicit when it comes to load-bearing behavior like this.

I would instead mention in security considerations that, unlike in regular CONNECT-UDP, the clients can now receive unwanted traffic from other hosts, and should be ready to handle it.